[CI] GeoBoundingBoxQueryGeoShapeIT testLimit2BoundingBox failing

Test is failing after #70850

**Build scan:**
https://gradle-enterprise.elastic.co/s/py3dzaodrbg2o/tests/:server:internalClusterTest/org.elasticsearch.search.geo.GeoBoundingBoxQueryGeoShapeIT/testLimit2BoundingBox

**Reproduction line:**
`./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.search.geo.GeoBoundingBoxQueryGeoShapeIT.testLimit2BoundingBox" -Dtests.seed=D3765D3D92140835 -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dtests.locale=ru -Dtests.timezone=America/Argentina/San_Juan -Druntime.java=11`

**Applicable branches:**
master

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.search.geo.GeoBoundingBoxQueryGeoShapeIT&tests.test=testLimit2BoundingBox

**Failure excerpt:**
```
org.elasticsearch.index.mapper.MapperParsingException: Failed to parse mapping: using deprecated parameters [strategy] in mapper [location] of type [geo_shape] is no longer allowed

  at __randomizedtesting.SeedInfo.seed([D3765D3D92140835:EF8E03E81CBDEDC3]:0)
  at org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:288)
  at org.elasticsearch.index.mapper.MapperService.mergeAndApplyMappings(MapperService.java:266)
  at org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:250)
  at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.updateIndexMappingsAndBuildSortOrder(MetadataCreateIndexService.java:983)
  at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.lambda$applyCreateIndexWithTemporaryService$2(MetadataCreateIndexService.java:408)
  at org.elasticsearch.indices.IndicesService.withTempIndexService(IndicesService.java:628)
  at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexWithTemporaryService(MetadataCreateIndexService.java:406)
  at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequestWithV1Templates(MetadataCreateIndexService.java:483)
  at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequest(MetadataCreateIndexService.java:369)
  at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequest(MetadataCreateIndexService.java:376)
  at org.elasticsearch.cluster.metadata.MetadataCreateIndexService$1.execute(MetadataCreateIndexService.java:293)
  at org.elasticsearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:48)
  at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:686)
  at org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:308)
  at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:203)
  at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:140)
  at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:139)
  at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:177)
  at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:669)
  at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:241)
  at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:204)
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
  at java.lang.Thread.run(Thread.java:834)

  Caused by: java.lang.IllegalArgumentException: using deprecated parameters [strategy] in mapper [location] of type [geo_shape] is no longer allowed

    at org.elasticsearch.index.mapper.GeoShapeFieldMapper.lambda$static$0(GeoShapeFieldMapper.java:150)
    at org.elasticsearch.index.mapper.ObjectMapper$TypeParser.parseProperties(ObjectMapper.java:318)
    at org.elasticsearch.index.mapper.ObjectMapper$TypeParser.parseObjectOrDocumentTypeProperties(ObjectMapper.java:237)
    at org.elasticsearch.index.mapper.RootObjectMapper$TypeParser.parse(RootObjectMapper.java:153)
    at org.elasticsearch.index.mapper.MappingParser.parse(MappingParser.java:102)
    at org.elasticsearch.index.mapper.MappingParser.parse(MappingParser.java:96)
    at org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:286)
    at org.elasticsearch.index.mapper.MapperService.mergeAndApplyMappings(MapperService.java:266)
    at org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:250)
    at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.updateIndexMappingsAndBuildSortOrder(MetadataCreateIndexService.java:983)
    at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.lambda$applyCreateIndexWithTemporaryService$2(MetadataCreateIndexService.java:408)
    at org.elasticsearch.indices.IndicesService.withTempIndexService(IndicesService.java:628)
    at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexWithTemporaryService(MetadataCreateIndexService.java:406)
    at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequestWithV1Templates(MetadataCreateIndexService.java:483)
    at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequest(MetadataCreateIndexService.java:369)
    at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequest(MetadataCreateIndexService.java:376)
    at org.elasticsearch.cluster.metadata.MetadataCreateIndexService$1.execute(MetadataCreateIndexService.java:293)
    at org.elasticsearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:48)
    at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:686)
    at org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:308)
    at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:203)
    at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:140)
    at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:139)
    at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:177)
    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:669)
    at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:241)
    at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:204)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
    at java.lang.Thread.run(Thread.java:834)

```

--- a/server/src/internalClusterTest/java/org/elasticsearch/search/geo/AbstractGeoBoundingBoxQueryIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/search/geo/AbstractGeoBoundingBoxQueryIT.java
@@ -38,12 +38,12 @@ abstract class AbstractGeoBoundingBoxQueryIT extends ESIntegTestCase {
         return false;
     }
 
-    public abstract XContentBuilder getMapping() throws IOException;
+    public abstract XContentBuilder getMapping(Version version) throws IOException;
 
     public void testSimpleBoundingBoxTest() throws Exception {
         Version version = VersionUtils.randomIndexCompatibleVersion(random());
         Settings settings = Settings.builder().put(IndexMetadata.SETTING_VERSION_CREATED, version).build();
-        XContentBuilder xContentBuilder = getMapping();
+        XContentBuilder xContentBuilder = getMapping(version);
         assertAcked(prepareCreate("test").setSettings(settings).setMapping(xContentBuilder));
         ensureGreen();
 
@@ -121,7 +121,7 @@ abstract class AbstractGeoBoundingBoxQueryIT extends ESIntegTestCase {
     public void testLimit2BoundingBox() throws Exception {
         Version version = VersionUtils.randomIndexCompatibleVersion(random());
         Settings settings = Settings.builder().put(IndexMetadata.SETTING_VERSION_CREATED, version).build();
-        XContentBuilder xContentBuilder = getMapping();
+        XContentBuilder xContentBuilder = getMapping(version);
         assertAcked(prepareCreate("test").setSettings(settings).setMapping(xContentBuilder));
         ensureGreen();
 
@@ -188,7 +188,7 @@ abstract class AbstractGeoBoundingBoxQueryIT extends ESIntegTestCase {
     public void testCompleteLonRange() throws Exception {
         Version version = VersionUtils.randomIndexCompatibleVersion(random());
         Settings settings = Settings.builder().put(IndexMetadata.SETTING_VERSION_CREATED, version).build();
-        XContentBuilder xContentBuilder = getMapping();
+        XContentBuilder xContentBuilder = getMapping(version);
         assertAcked(prepareCreate("test").setSettings(settings).setMapping(xContentBuilder));
         ensureGreen();
 

--- a/server/src/internalClusterTest/java/org/elasticsearch/search/geo/GeoBoundingBoxQueryGeoPointIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/search/geo/GeoBoundingBoxQueryGeoPointIT.java
@@ -8,6 +8,7 @@
 
 package org.elasticsearch.search.geo;
 
+import org.elasticsearch.Version;
 import org.elasticsearch.common.xcontent.XContentBuilder;
 import org.elasticsearch.common.xcontent.XContentFactory;
 
@@ -16,7 +17,7 @@ import java.io.IOException;
 public class GeoBoundingBoxQueryGeoPointIT extends AbstractGeoBoundingBoxQueryIT {
 
     @Override
-    public XContentBuilder getMapping() throws IOException {
+    public XContentBuilder getMapping(Version version) throws IOException {
         XContentBuilder xContentBuilder = XContentFactory.jsonBuilder().startObject().startObject("_doc")
             .startObject("properties").startObject("location").field("type", "geo_point");
         xContentBuilder.endObject().endObject().endObject().endObject();

--- a/server/src/internalClusterTest/java/org/elasticsearch/search/geo/GeoBoundingBoxQueryGeoShapeIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/search/geo/GeoBoundingBoxQueryGeoShapeIT.java
@@ -8,20 +8,19 @@
 
 package org.elasticsearch.search.geo;
 
-import org.apache.lucene.util.LuceneTestCase;
+import org.elasticsearch.Version;
 import org.elasticsearch.common.xcontent.XContentBuilder;
 import org.elasticsearch.common.xcontent.XContentFactory;
 
 import java.io.IOException;
 
-@LuceneTestCase.AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/72602")
 public class GeoBoundingBoxQueryGeoShapeIT extends AbstractGeoBoundingBoxQueryIT {
 
     @Override
-    public XContentBuilder getMapping() throws IOException {
+    public XContentBuilder getMapping(Version version) throws IOException {
         XContentBuilder xContentBuilder = XContentFactory.jsonBuilder().startObject().startObject("_doc")
             .startObject("properties").startObject("location").field("type", "geo_shape");
-        if (randomBoolean()) {
+        if (version.before(Version.V_8_0_0) && randomBoolean()) {
             xContentBuilder.field("strategy", "recursive");
         }
         xContentBuilder.endObject().endObject().endObject().endObject();

